### ISSUE TITLE
TypedDict bug

### ISSUE THREAD

**Issue by pmantica1** [python/mypy#8457]
* Are you reporting a bug, or opening a feature request?
Bug with the TypedDict class.
* Please insert below the code you are checking with mypy.
```
from typing import cast, TypedDict


TestDict = TypedDict("TestDict", {"val1": int, "val2": str})

instance = {}
instance["val1"] = 1
instance["val2"] = "1"

cast(TestDict, instance)
```
* What is the behavior/output you expect?
I expected no mypy errors. I get the following error instead:
```weird_linting_case.py:8: error: Incompatible types in assignment (expression has type "str", target has type "int")```

* What are the versions of mypy and Python you are using?
I am using python 3.8 and mypy version 0.761. Using the current mypy version in the master branch did not resolve the issue.

If you remove either one of the keys from the dict the problem goes away. My guess is that mypy "sees" the first type and expects all the keys in the dict to have that same type for some reason.

**Comment by obi1kenobi** [comment]
`instance` is almost certainly inferred to be a `Dict[str, int]` when declared, `mypy` isn't able to guess that `instance` is meant to actually be a `TestDict`. You can use `reveal_type` to double-check: https://mypy.readthedocs.io/en/stable/common_issues.html#displaying-the-type-of-an-expression

If this is the case, this is not a bug in mypy — it is a bug in the code you are type-checking.

**Comment by pmantica1** [comment]
If I try to write:
```
instance: TestDict = {}
```
I get the following error:
```
weird_linting_case.py:6: error: Keys ('val1', 'val2') missing for TypedDict "TestDict"
```
(Line 6 refers to the ```instance: TestDict = {}``` line)

Does this mean that when creating a TypedDict you have to initialize it will all the keys at once?

**Comment by obi1kenobi** [comment]
Yes, all the keys have to be present at creation time, unless `total=False`: https://www.python.org/dev/peps/pep-0589/#totality

While I understand this may be frustrating in your use case, the amount of work mypy would have to do to avoid this situation is probably quite extreme.
